experimental/algorithm/LAGraph_RegularReachability.c

```diff
@@ -0,0 +1,329 @@
+                                         // the NFA
+
+    GrB_Index rows = 0 ;                 // utility matrix row count
+    GrB_Index cols = 0 ;                 // utility matrix column count
+    GrB_Index vals = 0 ;                 // utility matrix value count
+
+    GrB_Matrix *A = NULL ;
+    GrB_Matrix *AT = NULL ;
+    GrB_Matrix *B = NULL ;
+    GrB_Matrix *BT = NULL ;
+
+    LG_ASSERT (reachable != NULL && G != NULL && R != NULL && S != NULL,
+        GrB_NULL_POINTER) ;
+    (*reachable) = NULL ;
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (G[i] == NULL) continue ;
+        LG_TRY (LAGraph_CheckGraph (G[i], msg)) ;
+    }
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (R[i] == NULL) continue ;
+        LG_TRY (LAGraph_CheckGraph (R[i], msg)) ;
+    }
+
+    LG_TRY (LAGraph_Malloc ((void **) &A, nl, sizeof (GrB_Matrix), msg)) ;
+    LG_TRY (LAGraph_Malloc ((void **) &AT, nl, sizeof (GrB_Matrix), msg)) ;
+
+    for (int32_t i = 0; i < nl; i++) 
+    {
+        if (G[i] == NULL) 
+        {
+            A[i] = NULL ;
+            continue ;
+        }
+
+        A[i] = G[i]->A;
+        if (G[i]->kind == LAGraph_ADJACENCY_UNDIRECTED ||
+            G[i]->is_symmetric_structure == LAGraph_TRUE)
+        {
+            AT[i] = A[i] ;
+        }
+        else
+        {
+            AT[i] = G[i]->AT ;
+            LG_ASSERT_MSG (AT[i] != NULL, LAGRAPH_NOT_CACHED,
+                "G[i]->AT is required") ;
+        }
+    }
+
+    LG_TRY (LAGraph_Malloc ((void **) &B, nl, sizeof (GrB_Matrix), msg)) ;
+    LG_TRY (LAGraph_Malloc ((void **) &BT, nl, sizeof (GrB_Matrix), msg)) ;
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (R[i] == NULL)
+        {
+            B[i] = NULL ;
+            continue ;
+        }
+
+        B[i] = R[i]->A;
+        if (R[i]->kind == LAGraph_ADJACENCY_UNDIRECTED ||
+            R[i]->is_symmetric_structure == LAGraph_TRUE)
+        {
+            BT[i] = B[i] ;
+        }
+        else
+        {
+            BT[i] = R[i]->AT ;
+            LG_ASSERT_MSG (BT[i] != NULL, LAGRAPH_NOT_CACHED,
+                "R[i]->AT is required") ;
+        }
+    }
+
+    for (int i = 0; i < nl; i++) 
+    {
+        if (A[i] == NULL) continue ;
+
+        GRB_TRY (GrB_Matrix_nrows (&ng, A[i])) ;
+        break ;
+    }
+
+    for (int i = 0; i < nl; i++) 
+    {
+        if (B[i] == NULL) continue ;
+
+        GRB_TRY (GrB_Matrix_nrows (&nr, B[i])) ;
+        break ;
+    }
+
+    // Check all the matricies in graph adjacency matrix decomposition are
+    // square and of the same dimensions
+    for (int i = 0; i < nl; i++) 
+    {
+        if (A[i] == NULL) continue ;
+
+        GRB_TRY (GrB_Matrix_nrows (&rows, A[i])) ;
+        GRB_TRY (GrB_Matrix_ncols (&cols, A[i])) ;
+
+        LG_ASSERT_MSG (rows == ng && cols == ng, LAGRAPH_NOT_CACHED,
+            "all the matricies in the graph adjacency matrix decomposition "
```

Comment:
Is it a correct style?
1. The first letter in a lower case 
2. Now dot at the end.

Reply:
I refer to the algorithms in the `src/`, e.g. [BFS](https://github.com/SparseLinearAlgebra/LAGraph/blob/afeb9dffca463fb966f1d9637986af55b04df461/src/algorithm/LG_BreadthFirstSearch_SSGrB.c#L79)